vendor/github.com/coreos/go-oidc/v3/oidc/oidc.go

```diff
@@ -147,13 +179,13 @@ func NewProvider(ctx context.Context, issuer string) (*Provider, error) {
 		}
 	}
 	return &Provider{
-		issuer:       p.Issuer,
+		issuer:       issuerURL,
 		authURL:      p.AuthURL,
 		tokenURL:     p.TokenURL,
 		userInfoURL:  p.UserInfoURL,
 		algorithms:   algs,
 		rawClaims:    body,
-		remoteKeySet: NewRemoteKeySet(ctx, p.JWKSURL),
+		remoteKeySet: NewRemoteKeySet(cloneContext(ctx), p.JWKSURL),
```

Comment:
huh, this cloneContext() call means canceling the context passed to NewProvider does nothing to interrupt in-flight requests for background keyset updates ... that's surprising